addons/website_sale/models/website.py

```diff
@@ -213,6 +213,9 @@ def _prepare_sale_order_values(self, partner, pricelist):
         company = self.company_id or pricelist.company_id
         if company:
             values['company_id'] = company.id
+            terms_and_conditions = self.env['ir.config_parameter'].sudo().get_param('sale.use_sale_note') and company.sale_note
+            if terms_and_conditions:
```

Comment:
But if this is `terms_and_conditions`, then you fall back on the default behavior, which can still be incorrect? For example, in case you defined Tersm & Conditions for Company 1 but not for Company 2?

Reply:
> But if this is `terms_and_conditions`, then you fall back on the default behavior, which can still be incorrect? For example, in case you defined Tersm & Conditions for Company 1 but not for Company 2?

I don't understand what you mean. Would you mind reformulating?
Is it about the `self.env['ir.config_parameter'].sudo().get_param('sale.use_sale_note')` that may be on the wrong company ?

Reply:
My bad, I reformulated. An example:
- Set main company of OdooBot to company 1, make company 2 allowed for OdooBot
- Set terms & conditions for company 1, leave them empty for company 2
- Set the website to company 2
- Buy any item on the webshop

Since there are no T&C for company 2, you fall back on the default behavior => the T&C of company 1 are selected.

Reply:
What do you think of this new version?